provider/openstack/networking.go

```diff
@@ -0,0 +1,249 @@
+// Copyright 2012, 2013 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package openstack
+
+import (
+	"sync"
+
+	"github.com/juju/errors"
+	"github.com/juju/utils"
+	"gopkg.in/goose.v1/neutron"
+	"gopkg.in/goose.v1/nova"
+
+	"github.com/juju/juju/instance"
+)
+
+// Networking is an interface providing networking-related operations
+// for an OpenStack Environ.
+type Networking interface {
+	// AllocatePublicIP allocates a public (floating) IP
+	// to the specified instance.
+	AllocatePublicIP(instance.Id) (*string, error)
```

Comment:
This is not quite the right interface for this (it should take a network id as well, or similar), but should be fine for now.